### DISCUSSION TITLE
request to add deepinfra api

### DISCUSSION THREAD

**Discussion by thecord** [LostRuins/lite.koboldai.net#34] [Ideas]
deepinfra is really fair price  1$ for 1 million token. 
https://deepinfra.com/pricing
can you add them please they have 70B models for llama70b and airoboros-l2-70b-gpt4-1.4.1
I wish you accept my request, I am not so good in coding
 but I have seen they are using **openai** library as it shown below
```python
import openai

stream = True # or False

# Point OpenAI client to our endpoint
openai.api_key = "<YOUR DEEPINFRA TOKEN: deepctl auth token>"
openai.api_base = "https://api.deepinfra.com/v1/openai"

MODEL_DI = "meta-llama/Llama-2-70b-chat-hf"
chat_completion = openai.ChatCompletion.create(
    model=MODEL_DI,
    messages=[{"role": "user", "content": "Hello world"}],
    stream=stream,
    max_tokens=100,
    # top_p=0.5,
)

if stream:
    # print the chat completion
    for event in chat_completion:
        print(event.choices)
else:
    print(chat_completion.choices[0].message.content)
    ```

appreciate for what you are giving to the community 👍

**Comment by LostRuins** [comment]
Seems like this is already supported, as they are using OpenAI compatible APIs

**Reply by thecord** [reply]
I did try that brother, 
the only issue I encountered is the model name as you know the list only shown openai models (gtp3 or gpt4) but they use llama-2 so I can't change **model name** after **filling my token** and **base URL**  
, because they using deferent model name 
look at the image please, and if it work, please guide me through 😊

![image](https://github.com/LostRuins/lite.koboldai.net/assets/15277172/60d1656d-aaf7-4b7c-8e87-9fc72ac65aa6)

**Reply by LostRuins** [reply]
okay, i'll add llama-2

**Comment by LostRuins** [comment]
I've added the ability to select custom models and force the chat completions api

**Reply by thecord** [reply]
Thank you for you support brother,
I am really appreciate that,
Just I couldn't bass the Model name with custom mode

model="jondurbin/airoboros-l2-70b-gpt4-1.4.1"

**Reply by LostRuins** [reply]
What error do you get? Did you try select the chat completions mode?

**Reply by thecord** [reply]
I tried them both, alwayes returned same massages.
this one brother 
Error: TypeError: Cannot read properties of undefined (reading 'message')

![image](https://github.com/LostRuins/lite.koboldai.net/assets/15277172/3056a565-bc11-4a94-932b-4ee758521986)

**Comment by thecord** [comment]
thank you so much **LostRuins**
 it's working perfectly. I am so glad you fulfill my request. please accept my gratitude
 most appreciate it.

**Reply by LostRuins** [reply]
you're welcome!
